File: scripts/view_video.py
import os
import sys
import webbrowser
from pathlib import Path

video_path = Path(__file__).resolve().parent.parent / "docs" / "OneShot_Task_Drawer_Compatibility_Fixed.mp4"

if not video_path.exists():
    print(f"Video file not found at: {video_path}", file=sys.stderr)
    sys.exit(1)

print(f"Opening demonstration video: {video_path}")
if sys.platform == "win32":
    os.startfile(str(video_path))
else:
    webbrowser.open(video_path.as_uri())
print("Video player launched successfully.")

File: scripts/view_video.mjs
import { exec } from 'node:child_process';
import { existsSync } from 'node:fs';
import { resolve } from 'node:path';

const videoPath = resolve(process.cwd(), 'docs/OneShot_Task_Drawer_Compatibility_Fixed.mp4');

if (!existsSync(videoPath)) {
  console.error(`Video file not found at: ${videoPath}`);
  process.exit(1);
}

console.log(`Opening demonstration video: ${videoPath}`);

const cmd = process.platform === 'win32'
  ? `start "" "${videoPath}"`
  : process.platform === 'darwin'
  ? `open "${videoPath}"`
  : `xdg-open "${videoPath}"`;

exec(cmd, (err) => {
  if (err) {
    console.error('Failed to open video automatically:', err.message);
  } else {
    console.log('Video player launched successfully.');
  }
});
